Source/Extensions/Array+Queueable.swift

```diff
@@ -0,0 +1,17 @@
+public protocol Queueable {
+  func process() -> Bool
+}
+
+public extension Array where Element : Queueable {
+
+  public mutating func processQueue(from: Int = 0, _ to: Int? = nil, process: ((object: Element) -> Void)? = nil) {
+    let to = to != nil ? to! : count
+    let currentQueue = self[from..<to]
+    for (index, queuable) in currentQueue.enumerate() {
```

Comment:
for (index, queuable) in currentQueue.enumerate() where queuable.process() {

}